pandas/core/internals/managers.py

```diff
@@ -817,6 +818,12 @@ def _make_na_block(
 
         if fill_value is None:
             fill_value = np.nan
+            dtype = interleaved_dtype([blk.dtype for blk in self.blocks])
+            if is_float_dtype(dtype):
+                # GH45857 avoid unnecessary upcasting
+                dtype = cast(np.dtype, dtype)
```

Comment:
is there a risk of getting e.g. Float64Dtype here?

Reply:
I don't think there is a risk in a DataFrame of only float32 values.

Reply:
right, im talking about a case where the DataFrame isnt all-float32, but includes at least one Float64Dtype